### ISSUE TITLE
Events doesnt work

### ISSUE THREAD

**Issue by dYb** [facebook/react#5229]
Is there anything can make click-event not work?
I`ve met a problem, every on-click event are not fired when i click it.
Those all happened on android webview, and just a few device have this problem.
thx for youre help

**Comment by mattstyles** [comment]
There's lots of potential culprits here

First eradicate some of the really obvious things that can sometimes trip you up, there's not an element grabbing clicks 'on top' of your intended target?

It's not a typo when adding the onClick handler in the jsx?

```
 return  <div onClick={ this.onClick }></div>
```

Maybe posting a code snippet of your component would help? Although I'm not sure a github issue is the best place for this sort of question

**Comment by dYb** [comment]
I tested it with react v 0.13.3 , all worked.
but when i change version to v0.13.3. it failed.
that happens on some android webview. phone model is Lenovo A820, android version is 4.1.2.

here is my code.

with version v0.13.3
```
import React from 'react';
class Root extends React.Component {
  constructor(props) {
      super(props);
      this.alert = this.alert.bind(this)
  }
  alert(){
    alert(123)
  }
  render(){
    let style = {
      'cursor': 'pointer'
    }
    return <div onClick={this.alert} style={style}>bbbbcc</div>
  }
}
React.render(<Root />,  document.getElementById('react-dom'))

```

with version 0.14
```
import React from 'react';
import ReactDOM from 'react-dom';
class Root extends React.Component {
  constructor(props) {
      super(props);
      this.alert = this.alert.bind(this)
  }
  alert(){
    alert(123)
  }
  render(){
    let style = {
      'cursor': 'pointer'
    }
    return <div onClick={this.alert} style={style}>bbbbcc</div>
  }
}
ReactDOM.render(<Root />,  document.getElementById('react-dom'))

```

**Comment by mattstyles** [comment]
If its on mobile it may be related to initialising touch events, as this was changed in 0.14

https://facebook.github.io/react/blog/2015/10/07/react-v0.14.html#breaking-changes

Although, I wasnt aware you had to do anything special, but, I guess the question is if there is anything else in your code that was required for 0.13.x that is now redundant but possibly screwing with something?

**Comment by dYb** [comment]
no, the above are all code. This is a test address http://f2e.developer.163.com/ybduan/game/index.html 
and you can check it on some cheap androlid mobiles

**Comment by dYb** [comment]
no, th above is all code.
There is a page for test.
http://f2e.developer.163.com/ybduan/game/index.html
You can check it on some cheap android mobiles, and use android origin webview.

**Comment by mattstyles** [comment]
I've only got access to an iphone, where things work.

I'm guessing you've tried to extract the React code and just try it with vanilla JS/DOM for that phone? Although that should obviously work, the android version is decent enough.

**Comment by dYb** [comment]
yes , i changed to v 0.13.3 it worked well.
And, on most moderner mobiles it worked too .

**Comment by mattstyles** [comment]
Good to see the wonderful world of Android is still providing endless joy for web developers!

You could try asking this question on the forums or on the reactiflux discord channel http://www.reactiflux.com/, seems pretty active and it only get more so as peeps move over from slack

**Comment by dYb** [comment]
> Good to see the wonderful world of Android is still providing endless joy for web developers!

haha.

Thank you very much . 谢谢。

**Comment by dYb** [comment]
A bad solution
```
class Like extends React.Component {
  constructor(props) {
    super(props);
    this.like = this.like.bind(this)
  }
  componentDidMount(){
    this.refs.dom.addEventListener('click', this.like, false)
  }
  componentWillUnmount(){
    this.refs.dom.removeEventListener('click', this.like, false)
  }
  like(){
    //...
  }
  render(){
    return (
      <div className="u-like" ref="dom">
        <span className="icon icon-2-01"></span>
      </div>
    )
  }
}
```

**Comment by mattstyles** [comment]
Hmm, that is odd, does `onClickCapture` work the 'normal' react way?

Just in case its some weird thing with capture phase on that device

**Comment by dYb** [comment]
It does`t work either.

**Comment by uncinimichel** [comment]
I had the same problem with IE.
So basically when the code is not boundled in 'production' mode, ReactErrorUtils will simulate the real browser event for better devtools logging.
You can go around the problem by boundling your code with process.env.NODE_ENV='production'

**Comment by dYb** [comment]
@uncinimichel  That`s it, it really works ,and the js files are smaller.

**Comment by uncinimichel** [comment]
@dYb that's good however we need to raise this a bug

**Comment by dYb** [comment]
@uncinimichel  I don`t think it is a bug of react, but it should be warded indeedly.

**Comment by zpao** [comment]
Switching to prod mode isn't a great long term solution. Luckily, we fixed this already and will ship that fix in 0.14.1 - see #5157

**Comment by uncinimichel** [comment]
@zpao That it is great! Thanks
